cs/src/core/Index/FASTER/FASTERThread.cs

```diff
@@ -180,8 +180,8 @@ internal bool InternalCompletePending(FasterExecutionContext ctx, bool wait = fa
 
                 if (wait)
                 {
-                    // Yield before checking again
-                    Thread.Yield();
+                    // SpinOnce before checking again
+                    new SpinWait().SpinOnce();
```

Comment:
I think SpinWait is intended to be reused outside the loop as it has state (count) which is used to decided between regular spin and yield.

Reply:
I moved it. Thank you.